src/gui/gui_nation_picker.hpp

```diff
@@ -215,6 +215,9 @@ class select_save_game : public button_element_base {
 			} else {
 				state.fill_unsaved_data();
 			}
+			/* Savefiles might load with new railroads, so for responsiveness we
```

Comment:
this is true, but it probably makes more sense to put this at the end of regenerate unsaved data, since you know that has to be called wherever loading is done